iroh-net/src/hp/derp/http/client.rs

```diff
@@ -214,13 +217,40 @@ impl ClientBuilder {
         }
     }
 
+    /// Build the derp http client with the expectation that we are dialing the
+    /// server using url.
+    pub fn build_with_server_url(self, key: key::node::SecretKey, url: Url) -> Client {
```

Comment:
Naming of this and `new_region` is a bit confusing. Would be nice to unify & clarify the different use cases a bit

Reply:
also seems like most of the code is shared between the two, maybe share that somehow?